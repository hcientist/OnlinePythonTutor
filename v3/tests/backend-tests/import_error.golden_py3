{
  "code": "# should NOT allow for any imports\nimport os\n\nos.system(\"echo security breach\")\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "exception_msg": "ImportError: os not found or not supported\nOnly these modules can be imported:\n  __future__, abc, array, bisect, calendar, cmath,\n  collections, copy, datetime, decimal, doctest, fractions,\n  functools, hashlib, heapq, io, itertools, json,\n  locale, math, operator, pickle, pprint, random,\n  re, string, time, types, typing, unittest\n\nYou can also try \"Python 3.6 with Anaconda (experimental)\",\nwhich is slower but lets you import many more modules.\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "event": "exception"
    }
  ]
}
