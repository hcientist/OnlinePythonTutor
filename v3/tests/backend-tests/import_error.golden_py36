{
  "code": "# should NOT allow for any imports\nimport os\n\nos.system(\"echo security breach\")\n",
  "trace": [
    {
      "line": 2,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 2,
      "event": "exception",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": "",
      "exception_msg": "ImportError: os not found or not supported\nOnly these modules can be imported:\n  __future__, abc, array, bisect, calendar, cmath,\n  collections, copy, datetime, decimal, doctest, fractions,\n  functools, hashlib, heapq, io, itertools, json,\n  locale, math, operator, pickle, pprint, random,\n  re, string, time, types, typing, unittest\n\nYou can also try \"Python 3.6 with Anaconda (experimental)\",\nwhich is slower but lets you import many more modules.\n"
    }
  ]
}
