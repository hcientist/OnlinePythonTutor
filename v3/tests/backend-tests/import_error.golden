{
  "code": "# should NOT allow for any imports\nimport os\n\nos.system(\"echo security breach\")\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "exception_msg": "ImportError: os not found or not supported\nOnly these modules can be imported:\n  StringIO, __future__, abc, array, bisect, calendar,\n  cmath, collections, copy, datetime, decimal, doctest,\n  fractions, functools, hashlib, heapq, io, itertools,\n  json, locale, math, operator, pickle, pprint,\n  random, re, string, time, types, unittest\n\nYou can also try \"Python 3.6 with Anaconda (experimental)\",\nwhich is slower but lets you import many more modules.\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "event": "exception"
    }
  ]
}
